src/codec/video/frame.rs

```diff
@@ -32,6 +32,43 @@ extern "C" {
     fn ffw_frame_free(frame: *mut c_void);
     fn ffw_frame_is_writable(frame: *const c_void) -> c_int;
     fn ffw_frame_make_writable(frame: *mut c_void) -> c_int;
+    fn ffw_frame_get_picture_type(frame: *const c_void) -> c_int;
+    fn ffw_frame_set_picture_type(frame: *mut c_void, picture_type: c_int);
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum PictureType {
+    None = 0,
+    /// Intra.
+    I = 1,
+    /// Predicted.
+    P = 2,
+    /// Bi-dir predicted.
+    B = 3,
+    /// S(GMC)-VOP MPEG4.
+    S = 4,
+    /// Switching Intra.
+    Si = 5,
+    /// Switching Predicted.
+    Sp = 6,
+    /// BI type.
+    Bi = 7,
+}
+
+impl PictureType {
+    pub fn from_i32(value: i32) -> Option<Self> {
+        match value {
+            0 => Some(PictureType::None),
+            1 => Some(PictureType::I),
+            2 => Some(PictureType::P),
+            3 => Some(PictureType::B),
+            4 => Some(PictureType::S),
+            5 => Some(PictureType::Si),
+            6 => Some(PictureType::Sp),
+            7 => Some(PictureType::Bi),
+            _ => None,
+        }
+    }
```

Comment:
1. I'd make this method available only within the crate itself. Let's not expose the implementation details.
2. I guess the `Option` is redundant here. I'd simply return `PictureType::None` in the undefined cases.
3. This method is being used to get the enum variant from a number returned by the C function. The C function returns `c_int` (which is fine) but here you expect `i32`. Do not confuse these two types. They are not necessarily the same thing on all platforms.
4. I'd name this method `from_raw` to remain consistent with the rest of the crate.

```suggestion
    pub(crate) fn from_raw(value: c_int) -> Self {
        match value {
            1 => PictureType::I,
            2 => PictureType::P,
            3 => PictureType::B,
            4 => PictureType::S,
            5 => PictureType::Si,
            6 => PictureType::Sp,
            7 => PictureType::Bi,
            _ => PictureType::None,
        }
    }
```